src/table/components/__tests__/TableBodyWrapper.spec.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { stardust } from '@nebula.js/stardust';
+import { render, fireEvent } from '@testing-library/react';
+import { generateDataPages, generateLayout } from '../../../__test__/generate-test-data';
+import manageData from '../../../handle-data';
+import { TableContextProvider } from '../../context';
+import TableBodyWrapper from '../TableBodyWrapper';
+import * as selectionsUtils from '../../utils/selections-utils';
+import * as getCellRenderer from '../../utils/get-cell-renderer';
+import * as handleKeyPress from '../../utils/handle-key-press';
+import * as handleAccessibility from '../../utils/handle-accessibility';
+import {
+  TableData,
+  ExtendedSelectionAPI,
+  TableLayout,
+  ExtendedTheme,
+  ChangeSortOrder,
+  PageInfo,
+  SetPageInfo,
+  ExtendedTranslator,
+  Cell,
+  CellHOC,
+} from '../../../types';
+
+describe('<TableBodyWrapper />', () => {
+  const rootElement = {} as HTMLElement;
+  const setShouldRefocus = () => undefined;
+  const keyboard = {} as stardust.Keyboard;
+  const tableWrapperRef = {} as React.MutableRefObject<HTMLDivElement | undefined>;
+  const announce = () => undefined;
+  const model = { getHyperCubeData: async () => generateDataPages(2, 2) } as unknown as EngineAPI.IGenericObject;
+
+  let tableData: TableData;
+  let constraints: stardust.Constraints;
+  let selectionsAPI: ExtendedSelectionAPI;
+  let layout: TableLayout;
+  let theme: ExtendedTheme;
+  let children: JSX.Element;
+  let tableFirstRow: Cell;
+  let tableSecondRow: Cell;
+
+  const renderTableBody = () =>
+    render(
+      <TableContextProvider selectionsAPI={selectionsAPI}>
+        <TableBodyWrapper
+          tableData={tableData}
+          constraints={constraints}
+          selectionsAPI={selectionsAPI}
+          layout={layout}
+          theme={theme}
+          rootElement={rootElement}
+          setShouldRefocus={setShouldRefocus}
+          keyboard={keyboard}
+          tableWrapperRef={tableWrapperRef}
+          announce={announce}
+        >
+          {children}
+        </TableBodyWrapper>
+      </TableContextProvider>
+    );
+
+  beforeEach(async () => {
+    jest.spyOn(selectionsUtils, 'addSelectionListeners').mockImplementation(() => jest.fn());
```

Comment:
```suggestion
    jest.spyOn(selectionsUtils, 'addSelectionListeners');
```

Reply:
There is a bunch of `api.on( ... )` lines in `addSelectionListeners` that by this change,  are not functioning, seems this line is needed.

Reply:
that makes sense actually. Try with
```suggestion
    jest.spyOn(selectionsUtils, 'addSelectionListeners').mockImplementation();
```